### DISCUSSION TITLE
What is the appropriate way to query a postgres `Date` type when using javascript dates?

### DISCUSSION THREAD

**Discussion by shohoney** [prisma/prisma#16905] [Q&A]
We have a field that we're storing as a `@db.Date` with Postgres.  We're trying to query for records where date field is less than or equal to the current date.  I'm getting future dates back in my result.  Example of the query
```typescript
const results = await prisma.model.findMany({
  where: {
    myPgDateField: {
       lte: new Date()
    }
});
```
The above is returning records where that date is a future date.  Am I missing something? I've tried setting the date object to UTC and several other things like setting the time to 0, but nothing seems to work.

**Comment by nurul3101** [comment]
Hello @shohoney 👋 

That's strange, Can you try suggestions mentioned in this GitHub Discussion and see if you get the same outcome?
- #11443

**Reply by shohoney** [reply]
I did try something similar, basically
```typescript
const date = new Date();
const dateToCompare = new Date(date.toISOString().slice(0, 10)); // the string passed in will be `YYYY-MM-DD`
const results = await prisma.model.findMany({
  where: {
    myPgDateField: {
       lte: dateToCompare
    }
});
```
which did not work.  I also threw a `// @ts-expect error` on this and tried using an ISO string like the on above and that also did not work.

**Reply by nurul3101** [reply]
> I'm getting future dates back in my result

Is the dates that you are getting a few hours ahead in future but have the same date, or you are getting many days ahead in future date? I am thinking if it's a few hours ahead in future but have the same date that you are passing then it could be an issue of time zone.

**Reply by shohoney** [reply]
They are literal Postgres dates, so they're stored using the @db.Date nativeType.  In the particular case of my test that I wrote, the date that was in the database vs the date that was being tested were 1 full year apart.

**Reply by pantharshit00** [reply]
Can you check which timezone setting your database is using? You can do so by following this stackoverflow thread: https://stackoverflow.com/questions/28216923/how-do-i-get-the-current-timezone-name-in-postgres-9-3

If it is not in UTC, it will have differences. Otherwise, can you provide a small reproduction in a git repo with a database dump which reproduces this problem?

**Reply by shohoney** [reply]
I can do that, but I have to ask how a date that is a full year different would be impacted in anyway by timezone before I move forward.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
